pkg/cosi/cosi_bucket_claim.go

```diff
@@ -0,0 +1,346 @@
+		util.Logger().Fatalf("❌ %s", err)
+	}
+	var b *nb.BucketInfo
+	if cosiBucketClaim.Status.BucketName != "" {
+		nbClient := sysClient.NBClient
+		bucket, err := nbClient.ReadBucketAPI(nb.ReadBucketParams{Name: cosiBucketClaim.Status.BucketName})
+		if err == nil {
+			b = &bucket
+		}
+	}
+
+	fmt.Printf("\n")
+	fmt.Printf("COSI BucketClaim info:\n")
+	fmt.Printf("  %-22s : %t\n", "Bucket Ready", cosiBucketClaim.Status.BucketReady)
+	fmt.Printf("  %-22s : kubectl get -n %s bucketclaim %s\n", "COSIBucketClaim", cosiBucketClaim.Namespace, cosiBucketClaim.Name)
+	fmt.Printf("  %-22s : kubectl get bucket %s\n", "COSIBucket", cosiBucket.Name)
+	fmt.Printf("  %-22s : kubectl get bucketclasses.objectstorage.k8s.io %s\n", "BucketClass", bucketClass.Name)
+	fmt.Printf("\n")
+
+	fmt.Printf("Shell commands:\n")
+	if b != nil {
+		fmt.Printf("Bucket status:\n")
+		fmt.Printf("  %-22s : %s\n", "Name", b.Name)
+		fmt.Printf("  %-22s : %s\n", "Type", b.BucketType)
+		fmt.Printf("  %-22s : %s\n", "Mode", b.Mode)
+		if b.PolicyModes != nil {
+			fmt.Printf("  %-22s : %s\n", "ResiliencyStatus", b.PolicyModes.ResiliencyStatus)
+			fmt.Printf("  %-22s : %s\n", "QuotaStatus", b.PolicyModes.QuotaStatus)
+		}
+		if b.Undeletable != "" {
+			fmt.Printf("  %-22s : %s\n", "Undeletable", b.Undeletable)
+		}
+		if b.NumObjects != nil {
+			fmt.Printf("  %-22s : %d\n", "Num Objects", b.NumObjects.Value)
+		}
+		if b.DataCapacity != nil {
+			fmt.Printf("  %-22s : %s\n", "Data Size", nb.BigIntToHumanBytes(b.DataCapacity.Size))
+			fmt.Printf("  %-22s : %s\n", "Data Size Reduced", nb.BigIntToHumanBytes(b.DataCapacity.SizeReduced))
+			fmt.Printf("  %-22s : %s\n", "Data Space Avail", nb.BigIntToHumanBytes(b.DataCapacity.AvailableSizeToUpload))
+			fmt.Printf("  %-22s : %s\n", "Num Objects Avail", b.DataCapacity.AvailableQuantityToUpload.ToString())
+		}
+		fmt.Printf("\n")
+	}
+}
+
+// RunListBucketClaim runs a CLI command
+func RunListBucketClaim(cmd *cobra.Command, args []string) {
+	list := &nbv1.COSIBucketClaimList{
+		TypeMeta: metav1.TypeMeta{Kind: "BucketClaim"},
+	}
+	if !util.KubeList(list) {
+		return
+	}
+	if len(list.Items) == 0 {
+		fmt.Printf("No COSI bucket claims found.\n")
+		return
+	}
+	table := (&util.PrintTable{}).AddRow(
+		"NAMESPACE",
+		"NAME",
+		"BUCKET-NAME",
+		"BUCKET-CLASS",
+		"BUCKET-READY",
+	)
+	for i := range list.Items {
+		cosiBucketClaim := &list.Items[i]
+		table.AddRow(
+			cosiBucketClaim.Namespace,
+			cosiBucketClaim.Name,
+			cosiBucketClaim.Status.BucketName,
+			cosiBucketClaim.Spec.BucketClassName,
+			fmt.Sprintf("%t", bool(cosiBucketClaim.Status.BucketReady)),
+		)
+	}
+	fmt.Print(table.String())
+}
+
+// WaitReady waits until the cosi bucket claim status bucket ready changes to true
+func WaitReady(cosiBucketClaim *nbv1.COSIBucketClaim) bool {
+	log := util.Logger()
+	klient := util.KubeClient()
+
+	intervalSec := time.Duration(3)
+
+	err := wait.PollImmediateInfinite(intervalSec*time.Second, func() (bool, error) {
```

Comment:
let's try to limit the time since we don't have a failed status like in obc.
in future PR please add a recording of error events so the user will know the failure reason.

Reply:
added retries